File: src/main/java/com/rimlang/rim/syntax/CommandDefinitionSyntaxNode.java
package com.rimlang.rim.syntax;

import com.rimlang.rim.lexer.StringToken;

import java.util.List;

public class CommandDefinitionSyntaxNode extends SyntaxNode {
    private final StringToken identifier;
    private final ArgsExpression args;
    private final List<SyntaxNode> body;

    public CommandDefinitionSyntaxNode(StringToken identifier, ArgsExpression args, List<SyntaxNode> body) {
        this.identifier = identifier;
        this.args = args;
        this.body = body;
    }

    public List<SyntaxNode> getBody() {
        return body;
    }

    public ArgsExpression getArgs() {
        return args;
    }

    public StringToken getIdentifier() {
        return identifier;
    }
}

File: src/main/java/com/rimlang/rim/syntax/node/global/CommandDefinitionSyntaxNode.kt
package com.rimlang.rim.syntax.node.global

import com.rimlang.rim.lexer.token.StringToken
import com.rimlang.rim.lexer.token.Token
import com.rimlang.rim.syntax.node.expression.ArgsExpression
import com.rimlang.rim.syntax.node.syntax.SyntaxNode

data class CommandDefinitionSyntaxNode(
    val identifier: StringToken,
    val args: ArgsExpression,
    val body: List<SyntaxNode>,
    override val firstToken: Token
) : SyntaxNode(firstToken)